File: 3754. Concatenate Non-Zero Digits and Multiply by Sum I/Solution.java
class Solution {
    public long sumAndMultiply(int n) {
        if (n == 0) return 0;

        int sum = 0;
        long concat = 0;
        long place = 1;

        while (n > 0) {
            int digit = n % 10;
            sum += digit;

            if (digit != 0) {
                concat = digit * place + concat;
                place *= 10;
            }

            n /= 10;
        }

        return concat * sum;
    }
}

File: 3754. Concatenate Non-Zero Digits and Multiply by Sum I/Solution.py
class Solution(object):
    def sumAndMultiply(self, n):
        """
        :type n: int
        :rtype: int
        """
        num = str(n)
        res = ""
        digitsum = 0

        for i in num:
            digit = int(i)
            digitsum += digit

            if digit!=0:
                res+=i  

        if res=="":
            return 0

        return int(res)*digitsum
